src/UbiTcp.cpp

```diff
@@ -0,0 +1,339 @@
+"Software"), to deal in the Software without restriction, including
+without limitation the rights to use, copy, modify, merge, publish,
+distribute, sublicense, and/or sell copies of the Software, and to
+permit persons to whom the Software is furnished to do so, subject to
+the following conditions:
+The above copyright notice and this permission notice shall be
+included in all copies or substantial portions of the Software.
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+Developed and maintained by Jose Garcia and Cristian Arrieta for IoT Services
+Inc
+@jotathebest at github: https://github.com/jotathebest
+@crisap94 at github: https://github.com/crisap94
+*/
+
+#include "UbiTcp.h"
+
+/**************************************************************************
+ * Overloaded constructors
+ ***************************************************************************/
+
+UbiTCP::UbiTCP(const char *host, const int port, const char *user_agent, const char *token) {
+  _host = host;
+  _user_agent = user_agent;
+  _token = token;
+  _port = port;
+  _syncronizeTime();
+}
+
+/**************************************************************************
+ * Destructor
+ ***************************************************************************/
+
+UbiTCP::~UbiTCP() {
+  delete[] _host;
+  delete[] _user_agent;
+  delete[] _token;
+}
+
+/**************************************************************************
+ * Cloud Functions
+ ***************************************************************************/
+
+bool UbiTCP::sendData(const char *device_label, const char *device_name, char *payload) {
+  bool allowed = _preConnectionChecks();
+  if (!allowed) {
+    return false;
+  }
+
+  /* Connecting the client */
+  _client_tcps_ubi.connect(_host, UBIDOTS_TCPS_PORT);
+  reconnect(_host, UBIDOTS_TCPS_PORT);
+  // if (!_client_tcps_ubi.verifyCertChain(_host)) {
+  //   if (_debug) {
+  //     Serial.println("[ERROR] Could not verify the remote secure server certificate, "
+  //                    "please make sure that you are using a secure "
+  //                    "network");
+  //   }
+  //   return false;
+  // }
+
+  /* Sends data to Ubidots */
+  if (_client_tcps_ubi.connected()) {
+    _client_tcps_ubi.print(payload);
+  } else {
+    if (_debug) {
+      Serial.println("[ERROR] Could not connect to the host");
+    }
+    _client_tcps_ubi.stop();
+    return false;
+  }
+
+  /* Waits for the host's answer */
+  if (!waitServerAnswer()) {
+    if (_debug) {
+      Serial.println("[ERROR] Could not read server's response");
```

Comment:
Missing PROGMEM allocation when printing the message `Serial.println(F("Message..."))`